File: frontend/types/recurso.ts
// Resource/exam-related TypeScript type definitions
export interface Recurso {
  id: number
  titulo: string
  tipo: "Examen" | "Practica" | "Silabo" | "PDF" | "Compendio" | "Libro" | "Apunte" | "Video"
  curso_id: number | null
  codigo_curso: string | null
  nombre_curso: string | null
  ciclo: number | null
  year: number | null
  downloads: number
  rating: number
  preview_url: string | null
  has_solucionario: boolean
  url_drive: string | null
  facultad_nombre: string | null
  created_at: string
}

File: backend/app/models/recurso.py
"""
Pydantic models for Resources
"""
from pydantic import BaseModel
from typing import Optional

class Recurso(BaseModel):
    id: int
    title: str
    code: str
    semester: str
    type: str  # "Examen", "Práctica", "Libro", "Apunte"
    ciclo: int
    facultad: str
    year: int
    downloads: int
    rating: float
    preview: bool
    hasSolucionario: bool
